app/src/test/java/org/mozilla/fenix/components/metrics/AppStartupTelemetryTest.kt

```diff
@@ -0,0 +1,163 @@
+package org.mozilla.fenix.components.metrics
+
+import android.content.Intent
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.verify
+import mozilla.components.support.utils.toSafeIntent
+import org.junit.Before
+import org.junit.Test
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Source.LINK
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Source.CUSTOM_TAB
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Source.APP_ICON
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Source.UNKNOWN
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Type.COLD
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Type.WARM
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Type.HOT
+
+class AppStartupTelemetryTest {
+
+    @RelaxedMockK
+    private lateinit var metricController: MetricController
+    @RelaxedMockK
+    private lateinit var intent: Intent
+
+    private lateinit var appStartupTelemetry: AppStartupTelemetry
+
+    @Before
+    fun setup() {
+        MockKAnnotations.init(this)
+        appStartupTelemetry = AppStartupTelemetry(metricController)
+    }
+
+    @Test
+    fun `WHEN application is launch for the first time through application icon THEN startup metric records {APP_ICON, COLD, FALSE}`() {
```

Comment:
Rather than saying `{APP_ICON, COLD, FALSE}`, which could go out of date if the implementation changes, you should probably use prose: `cold homescreen launch with no instance state`. That way if we refactor to send two probes, you could still potentially use the test without changing the name.

Honestly, you could also just say, "THEN records the correct values" because the front half of the test name is redundant to the back half. I would probably do this one.

Reply:
done!